Java/core/src/main/java/com/amazon/randomcutforest/returntypes/SampleSummary.java

```diff
@@ -156,21 +139,25 @@ public SampleSummary(List<Weighted<float[]>> points) {
      */
     public SampleSummary(List<Weighted<float[]>> points, double percentile) {
         checkArgument(points.size() > 0, "point list cannot be empty");
-        checkArgument(percentile > 0.5 && percentile < 1.0, "invalid ranges of the bound");
+        checkArgument(percentile > 0.5, " has to be more than 0.5");
+        checkArgument(percentile < 1.0, "has to be less than 1");
         int dimension = points.get(0).index.length;
         double[] coordinateSum = new double[dimension];
         double[] coordinateSumSquare = new double[dimension];
         double totalWeight = 0;
         for (Weighted<float[]> e : points) {
             checkArgument(e.index.length == dimension, "points have to be of same length");
             float weight = e.weight;
+            checkArgument(!Float.isNaN(weight), " weights must be non-NaN values ");
+            checkArgument(Float.isFinite(weight), " weights must be finite ");
             checkArgument(weight >= 0, "weights have to be non-negative");
             totalWeight += weight;
             for (int i = 0; i < dimension; i++) {
-                checkArgument(!Float.isNaN(weight) && Float.isFinite(weight),
-                        " weights must be finite, non-NaN values ");
-                checkArgument(!Float.isNaN(e.index[i]) && Float.isFinite(e.index[i]),
-                        " improper input, in coordinate " + i + ", must be finite, non-NaN values");
+                int index = i;
+                checkArgument(!Float.isNaN(e.index[i]),
+                        () -> " improper input, in coordinate " + index + ", must be non-NaN values");
+                checkArgument(Float.isFinite(e.index[i]),
+                        () -> " improper input, in coordinate " + index + ", must be finite values");
```

Comment:
Good callout. I was being reflexive -- I think the only check that should exist then is Float.isFinite(). RCF code cannot deal with infinities (it would need symbolic manipulation in any case). The weights should be non-negative, but all of the input numbers should be finite. That being said, for historical reasons, the core package did not test it everywhere. We should fix that -- but perhaps not in this go-around.